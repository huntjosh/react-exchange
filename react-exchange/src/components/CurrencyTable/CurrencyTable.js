import React, { Component } from 'react';
import PropTypes from 'prop-types';

class CurrencyTable extends Component {
  render() {
    return (
      <div>
        CurrencyTable
      </div>
    );
  }
}

CurrencyTable.propTypes = {};

export default CurrencyTable;
