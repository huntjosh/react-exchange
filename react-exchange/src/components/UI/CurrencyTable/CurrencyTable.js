import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/styles';
import Rate from '../../../models/Rate';
import CurrencyRow from '../CurrencyRow/CurrencyRow';

const styles = () => ({
  centered: {
    width: '100%',
    display: 'flex',
    justifyContent: 'center',
  },
});

// Here we will use PureComponent, as we don't have state,
// but we do want to avoid re-renders if the rates are the same
class CurrencyTable extends PureComponent {
  render() {
    const { rates, classes } = this.props;

    if (rates.length === 0) {
      return (
        <div className={classes.centered}>No Rates!</div>
      );
    }

    return (
      <React.Fragment>
        {rates.map(rate => (
          // We are relying on label + value being unique, we could set up a UUID
          <CurrencyRow rate={rate} key={`${rate.label}_${rate.value}`} />
        ))}
      </React.Fragment>
    );
  }
}

CurrencyTable.propTypes = {
  rates: PropTypes.arrayOf(Rate.propTypeStructure),
  // Override Because we define this object in this file
  classes: PropTypes.object.isRequired, // eslint-disable-line react/forbid-prop-types
};

CurrencyTable.defaultProps = {
  rates: [],
};

export default withStyles(styles)(CurrencyTable);
