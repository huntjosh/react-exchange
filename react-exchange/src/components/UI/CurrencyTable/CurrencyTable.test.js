import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import CurrencyTable from './CurrencyTable';

configure({ adapter: new Adapter() });
test('should render correctly with no props', () => {
  const component = shallow(<CurrencyTable />);
  expect(component).toMatchSnapshot();
});

test('should render currency table correctly with given rates', () => {
  const rates = [
    {
      label: '2018-08-18',
      value: 1.2388,
    },
    {
      label: '2018-08-17',
      value: 1.250,
    },
    {
      label: '2018-08-16',
      value: 1.267,
    },
    {
      label: '2018-08-15',
      value: 1.256,
    },
  ];

  const component = shallow(<CurrencyTable list={rates} />);
  expect(component).toMatchSnapshot();
});
