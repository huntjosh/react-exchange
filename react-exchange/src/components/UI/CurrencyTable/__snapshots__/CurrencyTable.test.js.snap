// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with no props 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyTable
    rates={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;

exports[`should render currency table correctly with given rates 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <CurrencyTable
    list={
      Array [
        Object {
          "label": "2018-08-18",
          "value": 1.2388,
        },
        Object {
          "label": "2018-08-17",
          "value": 1.25,
        },
        Object {
          "label": "2018-08-16",
          "value": 1.267,
        },
        Object {
          "label": "2018-08-15",
          "value": 1.256,
        },
      ]
    }
    rates={Array []}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": Symbol(react.fragment),
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": Symbol(react.fragment),
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
